### PR TITLE
Update `vantage-go` calls

### PR DESCRIPTION
**Pull request by conner**
swagger generation change in API. Most consequentially, the curated names given to endpoints in the API are now used in the generated swagger spec.

Also adds optional `VANTAGE_HOST` override to `make generate`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/hashicorp/terraform-plugin-docs v0.14.1
 	github.com/hashicorp/terraform-plugin-framework v1.5.0
 	github.com/hashicorp/terraform-plugin-testing v1.6.0
-	github.com/vantage-sh/vantage-go v0.0.12
+	github.com/vantage-sh/vantage-go v0.0.14-0.20240406212905-64e885583c48
```
Comment (whereandy) [inline go.mod:13]: eh?

Comment (conner) [reply go.mod:13]: once upstream stuff is merged + released, will re-point

(see https://github.com/vantage-sh/vantage-go/pull/12)

File: go.mod [hunk 2]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/hashicorp/terraform-plugin-docs v0.14.1
 	github.com/hashicorp/terraform-plugin-framework v1.5.0
 	github.com/hashicorp/terraform-plugin-testing v1.6.0
-	github.com/vantage-sh/vantage-go v0.0.13
+	github.com/vantage-sh/vantage-go v0.0.14-0.20240408182501-72e700aa1c5c
```
Comment (conner) [inline go.mod:13]: pseudo release via `go get github.com/vantage-sh/vantage-go@vntg-api-swagger-upgrade` to be replaced by actual release before merge.

**Review by whereandy — APPROVED** [review]
pending upstream changes

**Pull request merged**